configuration/pih/petl/jobs/hiv/viral_load/source.sql

```diff
@@ -0,0 +1,122 @@
+
+-- date encounter was created
+UPDATE temp_hiv_construct_encounters tvl JOIN encounter e ON tvl.encounter_id = e.encounter_id
+SET	tvl.date_created = e.date_created;
+
+## Delete test patients
+DELETE FROM temp_hiv_construct_encounters WHERE
+patient_id IN (
+               SELECT
+                      a.person_id
+                      FROM person_attribute a
+                      INNER JOIN person_attribute_type t ON a.person_attribute_type_id = t.person_attribute_type_id
+                      AND a.value = 'true' AND t.name = 'Test Patient'
+               );
+
+-- is specimen collection date estimated
+UPDATE temp_hiv_construct_encounters SET vl_sample_taken_date_estimated =  OBS_VALUE_CODED_LIST(encounter_id, 'PIH', '11781', 'en');
+
+-- lab result date
+UPDATE temp_hiv_construct_encounters SET vl_result_date =  OBS_VALUE_DATETIME(encounter_id, 'PIH', 'DATE OF LABORATORY TEST');
+
+-- viral load results (detectable)
+UPDATE temp_hiv_construct_encounters SET vl_result_detectable =  OBS_VALUE_CODED_LIST(encounter_id, 'CIEL', '1305', 'en');
+
+-- viral load results (numeric)
+UPDATE temp_hiv_construct_encounters SET viral_load =  OBS_VALUE_NUMERIC(encounter_id, 'CIEL', '856');
+
+-- The indexes are calculated using the specimen collection date
+### index ascending
+CREATE TEMPORARY TABLE temp_vl_index_asc
+(
+    SELECT
+            patient_id,
+            vl_sample_taken_date,
+            date_created,
+            encounter_id,
+            index_asc
+FROM (SELECT
+            @r:= IF(@u = patient_id, @r + 1,1) index_asc,
+            vl_sample_taken_date,
+            date_created,
+            encounter_id,
+            patient_id,
+            @u:= patient_id
+      FROM temp_hiv_construct_encounters,
+                    (SELECT @r:= 1) AS r,
+                    (SELECT @u:= 0) AS u
+            ORDER BY patient_id, vl_sample_taken_date ASC, date_created ASC
+        ) index_ascending );
+
+### index descending
+CREATE TEMPORARY TABLE temp_vl_index_desc
+(
+    SELECT
+            patient_id,
+            vl_sample_taken_date,
+            date_created,
+            encounter_id,
+            index_desc
+FROM (SELECT
+            @r:= IF(@u = patient_id, @r + 1,1) index_desc,
+            vl_sample_taken_date,
+            date_created,
+            encounter_id,
+            patient_id,
+            @u:= patient_id
+      FROM temp_hiv_construct_encounters,
+                    (SELECT @r:= 1) AS r,
+                    (SELECT @u:= 0) AS u
+            ORDER BY patient_id, vl_sample_taken_date DESC, date_created DESC
+        ) index_descending );
+
+### Final query
+SELECT
+        tvl.patient_id,
+        tvl.encounter_id,
+        DATE(tvl.vl_sample_taken_date),
+        vl_sample_taken_date_estimated,
+        vl_result_date,
+        vl_test_outcome,
+        vl_result_detectable,
+        viral_load,
+        vl_type,
+        DATEDIFF(NOW(), tvl.vl_sample_taken_date) days_since_vl,
```

Comment:
I prefer this much better to than the "days since last viral"!  Just want to make sure we confirm this change with the MEQ team though to avoid misinterpretation down the line.  (We should lobby hard for this way though IMO!)